osutil/cp_linux_test.go

```diff
@@ -36,11 +36,25 @@ func (s *cpSuite) TestCpMulti(c *C) {
 	c.Check(s.f2, testutil.FileEquals, s.data)
 }
 
+func (s *cpSuite) getReadOnlyFd(c *C) *os.File {
+	c.Check(osutil.CopyFile(s.f1, s.f2, osutil.CopyFlagDefault), IsNil)
+	c.Check(os.Chmod(s.f2, 0444), IsNil)
+	f2, err := os.Open(s.f2)
+	c.Assert(err, IsNil)
+	return f2
+}
+
 func (s *cpSuite) TestDoCpErr(c *C) {
-	f1, err := os.Open(s.f1)
+	src, err := os.Open(s.f1)
 	c.Assert(err, IsNil)
-	st, err := f1.Stat()
+	defer func() { src.Close() }()
+
+	st, err := src.Stat()
 	c.Assert(err, IsNil)
+
+	dstRo := s.getReadOnlyFd(c)
```

Comment:
(nitpick) I personally would not move this into a helper, given that it's only used once and is already super small. This way reading the test feels more like a "story" to me, with the function I need to jump to the function, understand it and jump back breaking my flow. But of course that is something very personal and once this is needed in multiple place extracting it would make sense.